File: lib/python/opentoken/src/main/opentoken/tokens/token_generation_exception.py
"""
Copyright (c) Truveta. All rights reserved.
"""


class TokenGenerationException(Exception):
    """Exception raised when token generation fails."""

    def __init__(self, message: str, cause: Exception = None):
        """
        Initialize the exception.

        Args:
            message: The error message.
            cause: The underlying cause of the exception.
        """
        super().__init__(message)
        self.cause = cause

File: lib/java/opentoken/src/main/java/com/truveta/opentoken/tokens/TokenGenerationException.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.tokens;

public class TokenGenerationException extends Exception {

    public TokenGenerationException(String message) {
        super(message);
    }

    public TokenGenerationException(String message, Throwable cause) {
        super(message, cause);
    }
}
